dashboard/app/hooks/useJob.ts

```diff
@@ -107,6 +108,28 @@ export default function useJobs(id: number | undefined = undefined) {
     return presignedUrl;
   }
 
+  async function authenticate(job: Job, password: string) {
+    try {
+      const authResponse = await fetch("api/job/authenticate", {
+      headers: new Headers({
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      }),
+      method: "POST",
+      body: JSON.stringify({
+        name: job.name,
+        password: password
+      })
+      });
+
+      if(!authResponse.ok) throw new Error("could not authenticate.")
+      
+      return true
+    }catch(error) {
```

Comment:
Ever heard about fixers? :D

Reply:
I'm not quite sure what you mean.

I added logging here, as it was missing.